modules/carto/src/layers/carto-layer.js

```diff
@@ -100,14 +103,31 @@ export default class CartoLayer extends CompositeLayer {
       let data;
 
       if (apiVersion === API_VERSIONS.V3) {
-        data = await getData({
+        const result = await getData({
           type,
           source,
           connection,
           credentials,
           geoColumn,
           columns
         });
+
+        if (result.format === FORMATS.NDJSON) {
+          const batches = await parseInBatches(result.data, NDJSONLoader, {
+            batchSize: 5000
+          });
+          const wait = r => setTimeout(r, 0);
+          const getDataAsync = async function* getDataAsync() {
+            for await (const batch of batches) {
+              // To avoid UI locking up, only yield once per frame
+              await new Promise(wait);
```

Comment:
Why do we need it?

Reply:
Well, it is optional. I just felt that without it the UI feels a bit sluggish when loading a large dataset. But it is of course a tradeoff, if it is removed the full dataset is displayed quicker. What do you think, which should we optimize for?